### PR TITLE
Refactor dialog

### PR DESCRIPTION
**Pull request by Konyaco**
Add basic `FluentDialog`, simple `ContentDialog`, and `FluentDialogHost` for command-style invocation.

### CODE REVIEW

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt [hunk 1]
```diff
@@ -64,63 +82,170 @@ fun Dialog(
     if (visibleState.currentState || visibleState.targetState) Popup(
         popupPositionProvider = DialogPopupPositionProvider
     ) {
+        val scrim by animateColorAsState(
+            if (visible) Color.Black.copy(0.3f) else Color.Transparent, animationSpec = tween(
```
Comment (Sanlorng) [inline fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt:86]: extend this as function
```Kotlin
private fun<T> dialogAnimationSpec() = tween<T>(
    easing = FluentEasing.FastInvokeEasing, 
    durationMillis = FluentDuration.ShotDuration
)
```
like `Flyout`

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt [hunk 2]
```diff
@@ -41,19 +47,31 @@ import com.konyaco.fluent.animation.FluentEasing
...
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.coroutines.resume
 
-internal expect val DialogPopupPositionProvider : PopupPositionProvider
+internal expect val DialogPopupPositionProvider: PopupPositionProvider
+
+@Stable
+class DialogSize(
```
Comment (Sanlorng) [inline fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt:59]: Rename to `DialogWidthRange`

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt [hunk 3]
```diff
@@ -64,63 +82,170 @@ fun Dialog(
...
+    onButtonClick: (ContentDialogButton) -> Unit,
+    size: DialogSize = DialogSize.Standard
+) {
+    FluentDialog(visible, size) {
+        Column {
+            Column(Modifier.background(FluentTheme.colors.background.layer.alt).padding(24.dp)) {
+                Text(
+                    style = FluentTheme.typography.subtitle,
+                    text = title,
+//                  color = FluentTheme.colors.text.text.primary
```
Comment (Sanlorng) [inline fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt:143]: Remove this comment

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt [hunk 4]
```diff
@@ -64,63 +82,170 @@ fun Dialog(
...
     }
-}
\ No newline at end of file
+}
+
+@Composable
+fun ContentDialogHost(state: ContentDialogHostState) {
+    val data = state.currentData
+
+    if (data != null) {
```
Comment (Sanlorng) [inline fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt:185]: Try to use a `DialogData.Empy` or Elvis operator﻿ to fix the issue of exit animation terminated

File: fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt [hunk 5]
```diff
@@ -64,63 +82,170 @@ fun Dialog(
...
+                ) {
+                    content()
+                }
+            }
+            // Divider
+            Box(Modifier.height(1.dp).background(FluentTheme.colors.stroke.surface.default))
+            // Button Grid
+            Box(Modifier.height(80.dp).padding(horizontal = 25.dp), Alignment.CenterEnd) {
+                Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
+                    AccentButton(modifier = Modifier.weight(1f), onClick = {
```
Comment (Sanlorng) [inline fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Dialog.kt:158]: format code from line `158` until `173`

**Comment by Konyaco** [comment]
image
image
image

Got some problems.. The previous workaround for removing 1px gap doesn't work anymore.  I tried to fix it, but got some inconsistent behavior, do you have any good idea?
See `1px_gap` branch

**Comment by Sanlorng** [comment]
> image image image
> 
> Got some problems.. The previous workaround for removing 1px gap doesn't work anymore. I tried to fix it, but got some inconsistent behavior, do you have any good idea? See `1px_gap` branch

could you provide the monitor configuration , content scale percent and gallery scale percent? i can't reprodue the problem in my monitor.

**Comment by Konyaco** [comment]
In any monitor and any scale. I accidentally commented here. Let's open an issue.

**Pull request merged**